website main menu: create support menu with community, wiki and commercial submenus

veroandreo opened the pull request:
This PR tries to bring about discussions about re-ordering menus and submenus. It's yet very raw, but the idea is to add a new menu "Support" with 3 children: "Community" (moved from About), "Wiki" (moved from Contribute) and a new "Commercial Support" page. In the latter, the idea would be to list companies or institutions that offer commercial support and services for GRASS (I put mundialis as an example). I used here the same layout as the download index page, so the idea would be to add company/institution logos, plus a description and the link in a button. See below: 

![image](https://github.com/OSGeo/grass-website/assets/20075188/81afaf85-799b-4660-9a71-dedd7d88eec4)

As layout for the Support index page, I copied the one from About menu:

![image](https://github.com/OSGeo/grass-website/assets/20075188/61c60b9c-ca2f-4dad-83d1-d9504446f02a)

Content needs to be added/edited and the layouts beautified, but this is a start point to get some feedback, so, please let me know what you think :)

nobeeakon commented on themes/grass/layouts/support/list.html:
```diff
@@ -0,0 +1,54 @@
+            </div>
+          </div>
+        </div>
+      </section>{{ partial "footer.html" . }}
```
weirdly footer.html has a closing </body></html> which turns this into invalid html, an option would be to move it at the end of the file

veroandreo replied:
mhm... I didn't change anything of that, just copied the layouts as they were and made minimal changes

nobeeakon commented on themes/grass/layouts/support/list.html:
```diff
@@ -0,0 +1,54 @@
+{{ "<!-- /navigation -->" | safeHTML }}
+
+
+    <section>
```
do we need this additional `<section>`? think is ok to remove

neteler commented:
Quick feedback: This looks great and I like a lot the splitting out of "Support" into an own section.

neteler commented:
The current conflicts need to be solved on command line...

neteler commented on themes/grass/layouts/support/commercial.html:
```diff
@@ -0,0 +1,82 @@
+
+            <div class="section bg-white mb-4">
+                <div class="row nm">
+                  <div class="col-4 text-center"><object type="image/svg+xml" data="{{.Site.BaseURL}}/images/logos/grasslogo.svg" class="grasslogo"></object></div>
+                  <div class="col-8">
+		    <h3 class="mt-20">mundialis GmbH & Co. KG</h3>
+		    <p>The GRASS GIS source code is available on <a href="https://github.com/OSGeo/grass" target="_blank"><i class="fa fa-github"></i> Github</a>, see all <a href="https://github.com/OSGeo/grass/tags">source code archives</a> for releases (zip and tarball files). Alternatively, find the latest tarballs and snapshots at <a href="https://grass.osgeo.org/grass{{- .Site.Data.grass.current_version_n ...
+            <a class="btn btn-primary" href="https://www.mundialis.de/en/" target="_blank">Visit our website</a>
```
The logo is available from here: https://data.neteler.org/tmp/mundialis_logo.svg

(I didn't manage to push into this PR)

neteler commented:
@veroandreo would you mind to fix the current merge conflict on command line?

veroandreo commented:
> @veroandreo would you mind to fix the current merge conflict on command line?

I only transferred the community page from one folder to another. How do I see these conflicts? 
The command line instructions provided here suggests me to create a new branch, test changes, merge with --no-ff... Are those right? They seem useful for testing someone else's PR

veroandreo commented:
So, I removed the wiki entry from Support menu:
![image](https://github.com/OSGeo/grass-website/assets/20075188/d6b49ae1-ae57-4ea3-b746-4555601570e9)

and added an icon to the support panel in home that appears because of this line: https://github.com/OSGeo/grass-website/blob/0bdb7170a1fcf7a12e7302b6261038672ef0bf3c/themes/grass/layouts/index.html#L36
![image](https://github.com/OSGeo/grass-website/assets/20075188/aa920603-f410-48e1-9786-4bdf375b6893)

Do we want 4 of those?

veroandreo commented:
fixed logo display
![image](https://github.com/OSGeo/grass-website/assets/20075188/c4628850-2909-41af-8dcf-e41b07caf7e5)

neteler commented:
I am fine with the 4 rather than 3 panels but do not visibly find the Wiki entry any more.

Indeed, it is under the photo on `support/community/` but please let's also (only?) have it in the text.

veroandreo commented:
> I am fine with the 4 rather than 3 panels but do not visibly find the Wiki entry any more.
> 
> Indeed, it is under the photo on `support/community/` but please let's also (only?) have it in the text.

Indeed. I removed the wiki entry from the support menu as we discussed in the last [PSC meeting](https://trac.osgeo.org/grass/wiki/PSC/Minutes/PSC_Meeting_20230811). 

Do you prefer to keep it as an entry in the support menu instead of the "panel" below the photo in community? IMHO, the Wiki might serve a number of purposes, i.e., contribute, find support, learn... where do you feel it matches better? Well, find support could equate learn too. Currently it is under contribute.

petrasovaa commented:
> Do you prefer to keep it as an entry in the support menu instead of the "panel" below the photo in community? IMHO, the Wiki might serve a number of purposes, i.e., contribute, find support, learn... where do you feel it matches better? Well, find support could equate learn too. Currently it is under contribute.

This goes into a discussion what the wiki is for. I think we should use it more as a community tool for planning and organizing rather than a place where people can learn or contribute given how easy wiki pages get obsolete. So I would keep wiki in Community, but I agree it deserves more than a picture, probably it could have it's own heading, because it doesn't quite fit into any of those categories on the community page.

petrasovaa commented:
@veroandreo I added various changes to this PR, it's ready for merge from my side. I changed the layout of the Support page and added short text, added NCSU CGA as commercial support, moved wiki from the panel to the page content within the community page and added fosstodon there. Is web.libera.chat still relevant?

neteler commented:
Would you mind to share a screenshot here?

petrasovaa commented:
![Selection_269](https://github.com/OSGeo/grass-website/assets/7494312/1fadb0ef-e882-47b1-a965-71b462f7475c)
![Selection_270](https://github.com/OSGeo/grass-website/assets/7494312/8727a213-ad8b-4ff3-b9e2-21241181128a)

veroandreo commented:
> @veroandreo I added various changes to this PR, it's ready for merge from my side. I changed the layout of the Support page and added short text, added NCSU CGA as commercial support, moved wiki from the panel to the page content within the community page and added fosstodon there. Is web.libera.chat still relevant?

Thanks for all the changes @petrasovaa ! I like them! Regarding libera chat, I logged in just out of curiosity and there were 2 others there, but no activity... I'd remove it.

veroandreo commented:
+1 for merging! :wink:

neteler commented:
Not what happened but the "Support" menu is entirely broken:

![image](https://github.com/OSGeo/grass-website/assets/1295172/f6430ee3-10e6-45e3-8e28-ac192ac2b305)


**404**:

- https://grass.osgeo.org/about/community/
- https://grass.osgeo.org/about/community/commercial
- https://grass.osgeo.org/about/community/community
